scripts/verify-commits.js

```diff
@@ -0,0 +1,42 @@
+const releaseTypes = ['major', 'minor', 'patch', 'prerelease'];
+let patterns = {
+  major: /<x\.[x|?]\.[x|?]>/,
+  minor: /<\?\.x\.[x|?]>/,
+  patch: /<\?\.\?\.x>/,
+  noRelease: /<[N|n][O|o]>/,
+};
+
+async function commitAnalyzer(pluginConfig, { options: { analyzeCommits }, commits, logger }) {
+  let releaseNumber = 3;
+  analyzeCommits.map((onePlugin) => {
+    patterns = {
```

Comment:
if this gets run multiple times, patterns will keep growning...

But perhaps more to the point, what does this do? Is the idea here that we can override the patterns in the config?

Reply:
This is to enable options to override patterns from package.json or .releaserc file, we could however we need to provide some default patterns. You are correct that this might keep growing, but this script is runned only once. And then release is triggered on non `master` branch this script isn't runned at all.